src/explore-education-statistics-common/src/modules/table-tool/utils/createExpandedRowHeaders.ts

```diff
@@ -91,31 +91,33 @@ function getRowHeadersInfo(rowHeaders: Header[]): {
 
   rowHeaders.forEach(header => {
     const { maxCrossSpan } = header;
+    const stack = [header];
 
-    let current: Header | undefined = header;
+    while (stack.length > 0) {
+      const current = stack.shift();
+
+      if (!current) {
+        break;
+      }
 
```

Comment:
Think we might be able to do a further optimisation here by checking if `!current.hasChildren`? e.g.

```ts
if (!current.hasChildren()) {
  continue;
}
```

Means we ignore the edge nodes, which never have a single matching child (and are hence always false for the collapsible level check).

Reply:
There's an eslint rule against using `continue`, it recommends using if statements instead so I've done that.